src/components/PreviewDraft/PreviewVersion.tsx

```diff
@@ -0,0 +1,166 @@
+  article: IArticle;
+  customTitle?: string;
+  language: string;
+}
+
+const SwitchWrapper = styled.div`
+  display: flex;
+  margin-right: ${spacing.xxlarge};
+  justify-content: flex-end;
+  align-items: center;
+  gap: ${spacing.xsmall};
+`;
+
+const TwoArticleWrapperWithDiff = styled(TwoArticleWrapper)`
+  del.diffmod,
+  ins.diffmod,
+  ins.mod {
+    background-color: ${colors.support.yellow};
+    text-decoration: none;
+    display: inline-block;
+  }
+
+  .diffins:has(img, div, figure, picture),
+  .diffmod:has(div, img, figure, picture) {
+    // Add some padding to show the diff outline on block elements
+    padding: 5px;
+  }
+
+  .diffins {
+    background-color: ${colors.support.greenLight};
+    text-decoration: none;
+    display: inline-block;
+  }
+
+  del.diffdel {
+    background-color: ${colors.support.redLight};
+    text-decoration: none;
+    display: inline-block;
+  }
+`;
+
+export const PreviewVersion = ({ article, language, customTitle }: VersionPreviewProps) => {
+  const [diffEnable, setDiffEnable] = useState(false);
+  const { t } = useTranslation();
+  const { values, initialValues } = useFormikContext<LearningResourceFormType>();
+  const { data: licenses = [] } = useLicenses();
+  const apiType = useMemo(
+    () => learningResourceFormTypeToDraftApiType(values, initialValues, licenses),
+    [initialValues, licenses, values],
+  );
+
+  const publishedArticle = toFormArticle(article, language);
+  const publishedTransformed = useTransformedArticle({ draft: publishedArticle, language });
+  const currentTransformed = useTransformedArticle({ draft: { ...apiType, id: article.id }, language });
+  const currentObj = useMemo(() => {
+    if (!diffEnable) return null;
+    return {
+      title: renderToString(currentTransformed.article?.title as ReactElement),
+      introduction: renderToString(currentTransformed.article?.introduction as ReactElement),
+      content: renderToString(currentTransformed.article?.content as ReactElement),
+    };
+  }, [diffEnable, currentTransformed]);
+  const publishObj = useMemo(() => {
+    if (!diffEnable) return null;
+    return {
+      title: renderToString(publishedTransformed.article?.title as ReactElement),
+      introduction: renderToString(publishedTransformed.article?.introduction as ReactElement),
+      content: renderToString(publishedTransformed.article?.content as ReactElement),
+    };
+  }, [diffEnable, publishedTransformed]);
+
+  const diffObj = useMemo(() => {
+    if (publishObj && currentObj) {
+      const titleDiff = getDiff(publishObj.title, currentObj.title);
+      const introductionDiff = getDiff(publishObj.introduction, currentObj.introduction);
+      const contentDiff = getDiff(publishObj.content, currentObj.content);
+      return { titleDiff, introductionDiff, contentDiff };
+    }
+    return null;
+  }, [currentObj, publishObj]);
+
+  const changeDiff = useCallback(() => {
+    setDiffEnable((p) => !p);
+  }, [setDiffEnable]);
```

Comment:
Her kunne vi kanskje spart oss for en useCallback

Reply:
Hvis du sier det, så tipper jeg det stemmer :+1: